arangod/Sharding/ShardingInfo.cpp

```diff
@@ -104,40 +103,16 @@ ShardingInfo::ShardingInfo(arangodb::velocypack::Slice info,
     }
   }
 
-  bool isASatellite = false;
-  auto replicationFactorSlice = info.get(StaticStrings::ReplicationFactor);
-  if (!replicationFactorSlice.isNone()) {
-    bool isError = true;
-    if (replicationFactorSlice.isNumber()) {
-      _replicationFactor = replicationFactorSlice.getNumber<size_t>();
-      // mop: only allow SatelliteCollections to be created explicitly
-      if (_replicationFactor > 0) {
-        isError = false;
-#ifdef USE_ENTERPRISE
-      } else if (_replicationFactor == 0) {
-        std::tie(isError, isASatellite) = makeSatellite();
-#endif
-      }
-    }
-#ifdef USE_ENTERPRISE
-    else if (replicationFactorSlice.isString() &&
-             replicationFactorSlice.copyString() == StaticStrings::Satellite) {
-      std::tie(isError, isASatellite) = makeSatellite();
-    }
-
-    if (isSmart && isASatellite) {
-      THROW_ARANGO_EXCEPTION_MESSAGE(
-          TRI_ERROR_BAD_PARAMETER,
-          "'isSmart' and replicationFactor 'satellite' cannot be combined");
-    }
-#endif
-    if (isError) {
-      THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER,
-                                     "invalid replicationFactor");
-    }
+  size_t replicationFactor = _replicationFactor;
+  Result res = extractReplicationFactor(info, isSmart, replicationFactor);
```

Comment:
can't just call ` Result res = extractReplicationFactor(info, isSmart, _replicationFactor);` without the new temp variable?

Reply:
In theory yes, but unfortunately the type of `_replicationFactor`  is `std::atomic<size_t>`, and not `size_t`, so I can't pass the member by reference here.